examples/WorkerExample/Readme.md

```diff
@@ -0,0 +1,140 @@
+    ```
+
+* Add a [Worker](./Worker.cs) class. Notes:
+    * It inherits from `BackgroundService`.
+    * It injects `TelemetryClient`.
+    * It sends one `request operation` and one `custom event` to application insights every 5 seconds.
+
+* Refer to [Program.cs](./Program.cs) for setup the worker as a generic host. Here are the critical services to register:
+
+    ```csharp
+    services.AddApplicationInsightsTelemetryWorkerService();    // Enable Application Insights for workers.
+    services.AddApplicationInsightsKubernetesEnricher();        // Enable Application Insights Kubernetes to enhance telemetries.
+    services.AddHostedService<Worker>();                        // Register the background service of Worker.
+    ```
+
+* Making sure `appsettings.json` will be copied to output folder on build, pay attention to the build task in the project file like this:
+    ```xml
+    <ItemGroup>
+        <!-- Build action is `None`, and the option for copy to output directory could be `Always` or `PreserveNewest` -->
+        <None Include="./appsettings.json" CopyToOutputDirectory="PreserveNewest" />
+    </ItemGroup>
+    ```
+
+    * Remember to update the connection string in `appsettings.json` when you try it on your own. You can get the connection string on the `Overview` blade of your application insights resource. Refer [Connection Strings](https://docs.microsoft.com/en-us/azure/azure-monitor/app/sdk-connection-string?tabs=net) for more details.
+    * Using of instrumentation key instead of connection string should work too. You will need to setup `InstrumentationKey` than `ConnectionString` in `appsettings.json`.
+
+## Build docker image
+
+* Build the image
+
+    ```shell
+    docker build -t workerapp .
+    ```
+
+* Optionally run it locally
+
+    ```shell
+    docker run -d --name myapp workerapp
+    docker logs myapp
+    ```
+
+    You will see logs like this:
+    
+    ```shell
+    info: Microsoft.Hosting.Lifetime[0]
+    Application started. Press Ctrl+C to shut down.
+    info: Microsoft.Hosting.Lifetime[0]
+        Hosting environment: Production
+    info: Microsoft.Hosting.Lifetime[0]
+        Content root path: /app
+    ```
+
+* Optionally, check out the `Live Metrics` in your application insights resource, you shall see request / custom events. 
+    * This is primary used to verify that the application is properly instrumented with Application Insights for worker.
+    * Since the application is not yet hosted in a Kubernetes cluster, there's no Kubernetes related property yet.
+
+        ![A screenshot showing request and custom events in live metrics](./images/LiveMetrics.png)
+
+* Delete the container when the test is done
+
+    ```shell
+    docker rm myapp -f
+    ```
+
+## Push the image when it is ready
+
+* Tag the image properly, for example:
+
+    ```shell
+    docker tag workerapp dockerhub_account_name/ai-k8s-worker-example:0.0.1
```

Comment:
Again, any registry that can be consumed from K8s should do, not just Docker Hub.